File: PassingCars/ruby/PassingCars.rb
class PassingCars

  def solution(a)

    west = 0
    east = 0
    
    a.each do |car|
        east += 1    if car==0
        west += east if car==1
    end
    
    return -1 if west > 1000000000
    west

  end

end

File: PassingCars/swift/PassingCars.swift
func solution(a: [Int]) -> Int {
    
    var east = 0
    var west = 0
    
    for car in a {
    
        if car==0 { east += 1 }
        if car==1 { west += east }
        
    }
    
    if (west > 1000000000) { return -1 }
    return west
    
}

println(solution(5,[0,1,0,1,1]))